lib/web_ui/lib/src/engine/window.dart

```diff
@@ -40,22 +41,35 @@ const int kImplicitViewId = 0;
 /// In addition to everything defined in [ui.FlutterView], this class adds
 /// a few web-specific properties.
 base class EngineFlutterView implements ui.FlutterView {
+  /// Creates a [ui.FlutterView] that can be used in multi-view mode.
+  ///
+  /// The [hostElement] parameter specifies the container in the DOM into which
+  /// the Flutter view will be rendered.
   factory EngineFlutterView(
     int viewId,
     EnginePlatformDispatcher platformDispatcher,
+    DomElement hostElement,
   ) = _EngineFlutterViewImpl;
 
   EngineFlutterView._(
     this.viewId,
     this.platformDispatcher,
-  );
+    // This is nullable to accommodate the legacy `EngineFlutterWindow`. In
+    // multi-view mode, the host element is required for each view (as reflected
+    // by the public `EngineFlutterView` constructor).
+    DomElement? hostElement,
+  )   : embeddingStrategy = EmbeddingStrategy.create(hostElement: hostElement),
+        _dimensionsProvider = DimensionsProvider.create(hostElement: hostElement);
```

Comment:
This is good for this review, but eventually the DimensionsProvider should be returned by the EmbeddingStrategy I think, they're slightly different "views" of the same `hostElement`.

Reply:
Agreed 👍